renovate/src/Renovate/Redirect/LayoutBlocks/Types.hs

```diff
@@ -92,63 +87,24 @@ data LayoutStrategy = LayoutStrategy
   , trampolines :: TrampolineStrategy
   } deriving (Eq, Ord, Read, Show)
 
-
-data Layout pair arch =
-  Layout { programBlockLayout :: [pair arch]
-         , layoutPaddingBlocks :: [ConcreteBlock arch]
+data Layout b arch =
+  Layout { programBlockLayout :: [WithProvenance b arch]
```

Comment:
Yadda yadda yadda I'm a fan of fewer type params and explicit/enforced relations between terms:
```
data Layout block =
  Layout { programBlockLayout :: [WithProvenance block]
         , layoutPaddingBlocks :: [PaddingBlock (ArchOf block)]
         , injectedBlockLayout :: [(SymbolicAddress (ArchOf block), ConcreteAddress (ArchOf block), BS.ByteString)]
         }
```